MigrationGuide.md

```diff
@@ -0,0 +1,335 @@
+# Guide for migrating to OpenAI 2.0.0-beta.1 from OpenAI 1.11.0
+
+This guide is intended to assist in the migration to the official OpenAI library (2.0.0-beta.1) from [OpenAI 1.11.0][openai_1110], focusing on side-by-side comparisons for similar operations between libraries. In this document, these are often referred to as the 'new' and the 'old' OpenAI libraries, respectively.
+
+Familiarity with the OpenAI 1.11.0 package is assumed. For those new to any OpenAI SDK for .NET, please refer to the [README][readme] rather than this guide.
+
+## Table of contents
+- [Client usage](#client-usage)
+- [Authentication](#authentication)
+- [Highlighted scenarios](#highlighted-scenarios)
+    - [Chat Completions: text generation](#chat-completions-text-generation)
+    - [Chat Completions: streaming](#chat-completions-streaming)
+    - [Chat Completions: JSON mode](#chat-completions-json-mode)
+    - [Chat Completions: vision](#chat-completions-vision)
+    - [Audio: speech-to-text](#audio-speech-to-text)
+    - [Audio: text-to-speech](#audio-text-to-speech)
+    - [Image: image generation](#image-image-generation)
+- [Additional samples](#additional-samples)
+
+## Client usage
+
+The client usage has considerably changed between libraries. While the old library had a single client, `OpenAIAPI`, from which multiple APIs could be accessed, the new library keeps a separate client per API. The following snippets illustrate this difference when invoking the image generation capability from the Image API:
+
+Old library:
+```cs
+OpenAIAPI api = new OpenAIAPI("<api-key>");
+ImageResult result = await api.ImageGenerations.CreateImageAsync("Draw a quick brown fox jumping over a lazy dog.", Model.DALLE3);
+```
+
+New library:
+```cs
+ImageClient client = new ImageClient("dall-e-3", "<api-key>");
+ClientResult<GeneratedImage> result = await client.GenerateImageAsync("Draw a quick brown fox jumping over a lazy dog.");
+```
+
+Another major difference highlighted in the snippets above is that the new library requires the model to be explicitly set during client instantiation, while the old client allows a model to be specified per call.
+
+The table below illustrates to which client each endpoint of `OpenAIAPI` was ported. Note that the deprecated Completions API is not supported by the new library:
+
+Old library's endpoint|New library's client
+|-|-
+|Chat | ChatClient
+|ImageGenerations | ImageClient
+|TextToSpeech | AudioClient
+|Transcriptions | AudioClient
+|Translations | AudioClient
+|Moderation | ModerationClient
+|Embeddings | EmbeddingClient
+|Files | FileClient
+|Models | ModelClient
+|Completions | Not supported
+
+## Authentication
+
+In order to authenticate to OpenAI you must set an API key when creating a client.
+
+The old library allowed setting the API key in three different ways: directly from a string, from an environment variable, or from a configuration file.
```

Comment:
```suggestion
The old library allowed setting the API key in 3 different ways:
- Directly from a string
- From an environment variable
- From a configuration file
```

Reply:
Couldn't commit the suggested changes because it got outdated. Just updated it manually.